### ISSUE TITLE
Questions about the Reported Results

### ISSUE THREAD

**Issue by adityalyer2000** [FengodChen/DMP-DUN-CVPR2025#2]
Hello, thank you for sharing your great work! I'm currently implementing DMP-DUN and have encountered some unusual things :

1. According to the reported results, compared with PRL-RND+ (Set11), DMP-DUN achieves an average PSNR improvement of approximately +1.12 dB, but its SSIM score is only about +0.006 higher. Why? In addition, there are noticeable seams visible where the image blocks meet. While these visual artifacts seem contradictory to the high reported PSNR values, my reproduction results are consistent with your reported data. 

2.  DMP-DUN+ (4-step) has a much larger parameter count and slower inference time than PRL-RND+, yet the reported FLOPs are much lower. In my own experiments with thop, the FLOPs for DMP-DUN+ (4-step) appear to be much higher than those of PRL-RND+.

Could you please help address my confusion? Thank you for your time and help!

**Comment by FengodChen** [comment]
Thank you very much for your attention and recognition of our work!

Regarding the first point: The occurrence of artifacts is normal because, due to GPU memory limitations, we set a small image cropping size during training—specifically, the cropping size is the same as the patch compression size. This prevents DMP-DUN from effectively handling differences between different patches. I believe this is also one of the reasons why the PSNR is higher while the SSIM is not, as PSNR only focuses on pixel-wise differences, while SSIM takes visual quality into account.

Regarding the second point: We also used thop to calculate the FLOPs, which should theoretically be consistent with the values in the table. I am not entirely sure what caused the discrepancy, but I suspect whether DMP-DUN was not set to eval mode before testing. In eval mode, the diffusion model in DMP-DUN automatically divides the image into small patches, which might affect the FLOPs calculation.

**Comment by adityalyer2000** [comment]
Thank you for your quick response. I calculated the FLOPs for DMP-DUN+ (4-steps) in eval mode using a (1, 1, 256, 256) random tensor, and the result was 4492.833 G, which seems surprisingly high! Could you take a look at my code below to see if I made a mistake?

````
from thop import profile  
x = torch.randn((1, 1, 256, 256))  
flops, params = profile(net, inputs=(x,))  
print("-" * 30)  
print(f"Total FLOPs: {flops / 1e9:.3f} G")  
print(f"Total Params: {params / 1e6:.3f} M")  
print("-" * 30)

**Comment by FengodChen** [comment]
Yes, following your approach, I did indeed get the same numbers. I apologize for my mistake earlier. At that time, I didn't calculate the FLOPs using thop, but instead referenced the FLOPs of ADM from other papers, then calculated the FLOPs of the modules I added to obtain the FLOPs for DMP-DUN-plus.

Unfortunately, I've forgotten which specific paper I referenced for the FLOPs at that time. After receiving your reply, I searched for a long time but couldn't find the original paper. Nevertheless, the FLOPs should not be as large as what was calculated using thop. For example, Table 2 in https://arxiv.org/pdf/2310.03337v3 lists the average GFLOPs per step of ADM (i.e., the U-Net model used in DMP-DUN) on CelebA-HQ (image size 256x256) as only 49.88.

**Comment by adityalyer2000** [comment]
Thank you for your time and help! I noticed that on page 6 of your recommended paper, the authors state that they conduct experiments on CelebA-HQ (64x64, 128x128), rather than 256x256. In addition, when calculating FLOPs for PRL-RND+ using thop, the result is significantly smaller. Is there anything wrong with calculating FLOPs via thop?

**Comment by FengodChen** [comment]
Oh, you are right. And maybe you could refer to the ImageNet results in Table 2, which report 82.06 GFLOPs per step. In short, what I’m trying to say is that, the actual FLOPs shouldn’t be as high as what thop measured.

Moreover, I downloaded the official PRL-PGD+ code and performed testing using thop. Since PRL employs the classic ResNet and U-shape architecture, the resulting FLOPs measurement is fairly accurate. In my environment, the model yields 561.38 GFLOPs. Below is my code, I hope it can be of some help to you! ^_^

```
import torch

from thop import profile
from model import PRL

cs_ratio = 0.1
B = 32
matrix_w = B**2
matrix_h = int(cs_ratio * matrix_w)

prl = PRL(
    nb = 7,
    B = B,
    Phi_init = torch.randn(matrix_h, matrix_w),
    nf = 16,
    ID_nf = 16
).eval()

with torch.no_grad():
    x = torch.randn(1, 1, 256, 256)
    macs, params = profile(prl, (x, ))

flops = macs * 2
gflops = flops / 1e9
print(f"GFLOPs: {gflops:.2f}")
```

**Comment by adityalyer2000** [comment]
Thank you for your response! The ImageNet results in Table 2 are for 64x64 resolution. As stated on page 6: "As for ImageNet, we aligned our processing with the unconditional ImageNet-64 settings as described in Nichol & Dhariwal (2021)."

I think using thop to calculate FLOPs is more rational than referencing reported FLOPs from other papers. Therefore, the FLOPs of DMP-DUN+ (4-step) are 8,985.66 G, and the FLOPs of PRL-RND+ are 561.38 G. This supports the observation that DMP-DUN+ (4-step) has a much larger parameter count and slower inference time than PRL-RND+.

**Comment by FengodChen** [comment]
Thank you for pointing out that error. However, there’s still one thing that confuses me. As stated on page 6 of the paper:

> For the CelebA-HQ-64, AFHQ, and LSUN-bedroom datasets, we adhered to the typical training data splits, normalizing and resizing the images to 64×64 pixels. As for ImageNet, we aligned our processing with the unconditional ImageNet-64 settings as described in Nichol & Dhariwal (2021).

If that’s the case, both CelebA-HQ and ImageNet should have images of size 64×64. But why does Table 2 show different GFLOPs for the two?

**Comment by adityalyer2000** [comment]
Thank you for your reply ! I think it's probably because they use the ADM model with different hyperparameters. Maybe ImageNet needs a large base channel number?

**Comment by FengodChen** [comment]
That makes sense. In any case, thank you very much for pointing out these error. I’ll correct it in a future version. ^_^
